src/helper/sampler.hpp

```diff
@@ -0,0 +1,84 @@
+// Unless explicitly stated otherwise all files in this repository are
+// dual-licensed under the Apache-2.0 License or BSD-3-Clause License.
+//
+// This product includes software developed at Datadog
+// (https://www.datadoghq.com/). Copyright 2022 Datadog, Inc.
+
+#include <iostream>
+#include <mutex>
+
+#pragma once
+
+namespace dds {
+class sampler {
+public:
+    sampler(bool enabled, double sample_rate) : enabled_(enabled)
```

Comment:
I think you should remove the enabled flag here and just use the enabled setting in the engine directly, the sampler shouldn't know whether the feature is enabled or not, it just needs to sample.

Reply:
Done